04_Scheduling/02_Scheduler.md

```diff
@@ -154,14 +154,64 @@ void schedule() {
 
 Now our schedule function every time that is called will pick the next not null item (task) in the array. But there is a problem: if there are no tasks in the array, the while loop will never end, and so the interrupt handler will never exit, leaving the kernel stuck there. And that's true, but we will see how to solve this problem later in the chapter, for now let's just pretend that this case never happen, and continue with our implementation.
 
-### Saving the task context 
+### Saving and restoring the task context 
 
-In the previous section we have seen how to iterate throught tasks, but that basically just iterating through items of a list, so for now our scheduler is still doing nothing. Once the next task to be executed has been selected selected, the first thing that the scheduler has to do is stop the current task, and save its current status. If you have implemented the interrupt handler following this guide you should already have the context being passed to the it, so now we need to change the signature of our schedule function, adding a new parameter: 
+In the previous section we have seen how to iterate throught tasks, but that basically was just iterating through items of a list, not very useful. 
+
+#### Saving Context
+
+Once the next task to be executed has been selected, the first thing that the scheduler has to do is stop the current task, and save its current status. If you have implemented the interrupt handler following this guide you should already have the context being passed to, what we want is to pass down the context to the schedule function, but first  we need to change its signature of our schedule function, adding a new parameter: 
```

Comment:
I would say the scheduler doesn't *stop* the current task, as it's already been paused when the interrupt fired. The task is not running in the background while the interrupt runs, the interrupt firing has already stopped the task.